api.yaml

```diff
@@ -84,6 +84,34 @@ paths:
             schema:
               $ref: '#/components/schemas/user'
         required: true
+  /users/{username}:
+    patch:
+      summary: Modifies the logged user, for exemple it's role.
+      tags:
+        - admins
+        - noroles
+      operationId: patchUser
+      description: |
+        Modifies the logged user, for exemple it new role to be added to the user
+      parameters:
+        - name: username
+          in: path
+          required: true
+          type: string
+          format: username
+          description: The username of a specific user
+        - name : JSONPatch
+          in: body
+          required: true
+          schema:
+              $ref: '#/components/schemas/user'
```

Comment:
Should be requestbody not a parameter in the body. In addition JSONPatch follows a specific syntax (RFC 6902) that is not required here. Basically, any object property (that is part of the object schema) should update the provided property value leaving the other properties intact.

Reply:
fixed